File: 단계별로 풀어보기/Step 18/11047.cpp
#include <iostream>
using namespace std;

int main() {
    int N, K;
    int coin[10];
    int i;
    int cnt = 0;

    cin >> N >> K;
    for(i=0; i<N; i++) {
        cin >> coin[i];
    }

    while(0 < K) {
        for(i=0; i<N; i++) {
            if(K < coin[i])
                break;
        }
        K -= coin[i-1];
        cnt++;
    }

    cout << cnt << endl;
    return 0;
}

File: StepByStep/Python/Step_18_그리디 알고리즘/11047.py
N, K = map(int, input().split())
coin = []
for i in range(N):
    coin.append(int(input()))

cnt = 0
for i in range(N-1, -1, -1):
    if K == 0:
        break
    if coin[i] <= K:
        cnt += K // coin[i]
        K = K % coin[i]

print(cnt)
